### PR TITLE
Unable to set email/username on a remote server

### PR DESCRIPTION
**Pull request by jzhang20133**
Fixes https://github.com/jupyterlab/jupyterlab-git/issues/1323

### CODE REVIEW

**Review by Zsailer — APPROVED** [review]

File: src/components/GitPanel.tsx
Comment (krassowski) [inline src/components/GitPanel.tsx:845]: The liner suggests to use triple equals here. The argument annotation says it cannot be undefined so it should be equivalent:


```suggestion
    if (path === null) {
```

Unless this fix depends on it, then we need to change linter eqeqeq rule to smart and add undefined in the argument.

Also, can you add a test?

**Review by krassowski — CHANGES_REQUESTED** [review]

**Comment by Zsailer** [comment]
Nice! Thanks @jzhang20133! 

This LGTM. @fcollonval I don't have access on this repo. Would you mind helping me merge and release a patch here? I can pick this up too if you want to give me access here.

**Comment by Zsailer** [comment]
> Also, can you add a test?

@krassowski I dived into this a bit. It appears that we're skipping most unit tests for this panel since the JupyterLab 4 release. (see [here](https://github.com/jupyterlab/jupyterlab-git/blob/b878c0b28376e7d847a095d6e5205596508421a5/src/__tests__/test-components/GitPanel.spec.tsx#L179), [here](https://github.com/jupyterlab/jupyterlab-git/blob/b878c0b28376e7d847a095d6e5205596508421a5/src/__tests__/test-components/GitPanel.spec.tsx#L215), [here](https://github.com/jupyterlab/jupyterlab-git/blob/b878c0b28376e7d847a095d6e5205596508421a5/src/__tests__/test-components/GitPanel.spec.tsx#L234), [here](https://github.com/jupyterlab/jupyterlab-git/blob/b878c0b28376e7d847a095d6e5205596508421a5/src/__tests__/test-components/GitPanel.spec.tsx#L250), and [here](https://github.com/jupyterlab/jupyterlab-git/blob/b878c0b28376e7d847a095d6e5205596508421a5/src/__tests__/test-components/GitPanel.spec.tsx#L266)). I believe this is the reason the current bug wasn't detected in the first place. Removing and of the `.skip` calls causes these tests to fail, so it seems that the API changed somewhere along the way and the tests were never updated? 

I think it's beyond the scope of this current PR to fix all of these failing unit tests. For the sake of addressing the current bug, I would propose that we merge the current PR without additional unit testing. 

Further, it seems that the workflows for this repo have been failing for quite awhile due to some dependency resolution errors.

**Comment by Zsailer** [comment]
I've opened the following issue to track work on re-enabling the skipped unit tests: https://github.com/jupyterlab/jupyterlab-git/issues/1325

**Comment by krassowski** [comment]
> I don't have access on this repo. Would you mind helping me merge and release a patch here? I can pick this up too if you want to give me access here.

I've added you as a collaborator, but let's get the CI green before merging - thank you for opening https://github.com/jupyterlab/jupyterlab-git/pull/1326!

**Comment by Zsailer** [comment]
> but let's get the CI green before merging 

Done, @krassowski 🚀 

I'll try to get the (skipped) unit tests working for the Git Panel. If I can't get that done today, I don't think we should block this PR.

**Comment by Zsailer** [comment]
Amazing work, @jzhang20133! Thank you so much for taking it the extra mile to fix all unit tests and adding a few new ones for this PR! 🚀

**Review by Zsailer — APPROVED** [review]

**Comment by Zsailer** [comment]
We'll need help with releasing this bug fix, since we don't have admin access to this repo. 

I think it would be worth updating the Github workflows for releasing to use the latest work happening in Jupyter-releaser with project level PyPI tokens.

**Comment by fcollonval** [comment]
Thanks a lot @jzhang20133 @Zsailer and @krassowski for the latest work; [v0.50.1 has been released](https://github.com/jupyterlab/jupyterlab-git/releases/tag/v0.50.1) with the latest fixes.

**Comment by Zsailer** [comment]
Thank you, @fcollonval! We appreciate you tremendously! :heart:

**Pull request merged**
